source/frontend/src/components/common/form/FileDragAndDrop.tsx

```diff
@@ -0,0 +1,152 @@
+import { ChangeEvent, FunctionComponent, useState } from 'react';
+import styled from 'styled-components';
+
+import { SectionField } from '../Section/SectionField';
+
+interface IFileDragAndDropProps {
+  onSelectFile: (file: File | null) => void;
+  selectedFile: File | null;
+  validExtensions: string[];
+}
+
+const FileDragAndDrop: FunctionComponent<React.PropsWithChildren<IFileDragAndDropProps>> = ({
+  onSelectFile,
+  selectedFile,
+  validExtensions,
+}) => {
+  const validDocumentExtensions: string = validExtensions.map(x => `.${x}`).join(',');
+
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleFileInput = (changeEvent: ChangeEvent<HTMLInputElement>) => {
+    // handle validations
+    if (changeEvent.target !== null) {
+      var target = changeEvent.target;
+      if (target.files !== null) {
```

Comment:
target.files?.length !== null & target.files.length >= 1
(or similar)

Reply:
sure, added the check for at least one file